Plan/src/main/java/com/djrapitops/plugin/VelocityPlugin.java

```diff
@@ -0,0 +1,101 @@
+package com.djrapitops.plugin;
+
+import com.djrapitops.plugin.api.Benchmark;
+import com.djrapitops.plugin.api.systems.TaskCenter;
+import com.djrapitops.plugin.api.utility.Version;
+import com.djrapitops.plugin.api.utility.log.DebugLog;
+import com.djrapitops.plugin.command.CommandNode;
+import com.djrapitops.plugin.command.velocity.VelocityCommand;
+import com.velocitypowered.api.proxy.ProxyServer;
+import net.kyori.text.TextComponent;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+
+/**
+ * (Based on BungeePlugin)
+ *
+ * @author MicleBrick
+ */
+public abstract class VelocityPlugin implements IPlugin {
+
+    protected boolean reloading;
+
+    @Override
+    public void onEnable() {
+        StaticHolder.register(this);
+    }
+
+    @Override
+    public void onDisable() {
+        Class<? extends IPlugin> pluginClass = getClass();
+        StaticHolder.unRegister(pluginClass);
+        Benchmark.pluginDisabled(pluginClass);
+        DebugLog.pluginDisabled(pluginClass);
+        TaskCenter.cancelAllKnownTasks(pluginClass);
+    }
+
+    @Override
+    public void reloadPlugin(boolean full) {
+        PluginCommon.reload(this, full);
+    }
+
+    @Override
+    public void log(String level, String s) {
+        Logger logger = getLogger();
+        switch (level.toUpperCase()) {
+            case "INFO":
+            case "I":
+                logger.info(s);
+                break;
+            case "INFO_COLOR":
+                getProxy().getConsoleCommandSource().sendMessage(TextComponent.of(s));
```

Comment:
`ComponentSerializers.LEGACY.deserialize(s)`

Reply:
Why does Velocity offer a legacy deserializer? What does this do?

Reply:
People still like to work with "legacy" chat types. Unless this actually prints ANSI color codes, I'd still stand by my recommendation.

Reply:
(It lets you use e.g. &e for yellow)